src/module/main.cpp

```diff
@@ -606,15 +661,11 @@ class Orbbec : public vsdk::Camera, public vsdk::Reconfigurable {
             uint64_t depthTS = depth->getSystemTimeStampUs();
             if (colorTS != depthTS) {
                 VIAM_SDK_LOG(info) << "color and depth timestamps differ, defaulting to "
-                                      "color";
-                VIAM_SDK_LOG(info) << "color timestamp was " << colorTS << "depth timestamp was " << depthTS;
-            }
-            uint64_t timestamp = colorTS == 0 ? depthTS : colorTS;
-
-            // throw if frame is older than a second.
-            if (getTimeSinceFrame(timestamp) > maxFrameAgeUs) {
-                throw std::invalid_argument("no recent frames: check USB connection");
+                                      "older of the two"
+                                   << "color timestamp was " << colorTS << " depth timestamp was " << depthTS;
             }
+            // use the older of the two timestamps
+            uint64_t timestamp = (colorTS > depthTS) ? depthTS : colorTS;
```

Comment:
would it be valuable to log both instead of taking the older?

Reply:
this isn't for logging. This is for which timestamp we use for the captured_at field. We can only pick one. If we have to choose, we should pick the oldest.